drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -120,21 +106,41 @@ static int siwx91x_ap_enable(const struct device *dev, struct wifi_connect_req_p
 		.is_11n_enabled      = 1,
 	};
 
+	interface = sl_wifi_get_default_interface();
+	if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) != SL_WIFI_AP_INTERFACE) {
+		LOG_ERR("Interface not in AP mode");
+		return -EINVAL;
+	}
+
+	if (sidev->state == WIFI_STATE_COMPLETED) {
+		return 0;
+	}
+
 	if (params->band != WIFI_FREQ_BAND_UNKNOWN && params->band != WIFI_FREQ_BAND_2_4_GHZ) {
+		LOG_ERR("Unsupported band");
 		return -ENOTSUP;
 	}
 
+	/*FIXME siwx91x_ap_cfg.channel.channel = SL_WIFI_BAND_2_4GHZ */
```

Comment:
I don't understand this `FIXME`. `SL_WIFI_BAND_2_4GHZ` does not make sense for `siwx91x_ap_cfg.channel.channel`. Do you mean `siwx91x_ap_cfg.channel.band` ?